gamestonk_terminal/portfolio/degiro_api.py

```diff
@@ -0,0 +1,484 @@
+        mask_not_empty = positions["size"] > 0
+        mask_current_position = mask_product & mask_not_empty
+
+        # FILTER
+        positions = positions[mask_current_position]
+
+        return positions
+
+    def __init__(self, credentials: Credentials):
+        self.__trading_api = TradingAPI(credentials=credentials)
+
+        self.__degiro_parser = argparse.ArgumentParser(
+            add_help=False,
+            prog="degiro",
+        )
+        self.__degiro_parser.add_argument("cmd", choices=self.CHOICES)
+
+    def q(self, _):
+        """Process Q command - quit the menu"""
+
+        return False
+
+    def quit(self, _):
+        """Process Quit command - quit the program"""
+
+        return True
+
+    def fetch_additional_information(
+        self,
+        positions: pd.DataFrame,
+    ) -> pd.DataFrame:
+        """Fetch extra information about the positions like :
+            - name
+            - isin
+            - symbol
+            - ...
+
+        Parameters
+        ----------
+        positions : pd.DataFrame
+            Positions from which we want extra fields.
+
+        Returns
+        -------
+        pd.DataFrame
+            Positions with additional data.
+        """
+
+        # GET ATTRIBUTES
+        trading_api = self.__trading_api
+
+        # EXTRACT POSITIONS IDS
+        positions_ids = positions["id"].astype("int32").tolist()
+
+        # FETCH EXTRA DATA
+        request = ProductsInfo.Request()
+        request.products.extend(positions_ids)
+        products_info_pb = trading_api.get_products_info(
+            request=request,
+            raw=False,
+        )
+
+        # CONVERT TO DICT
+        products_info_dict = pb_handler.message_to_dict(
+            message=products_info_pb,
+        )
+
+        # CONVERT TO DATAFRAME
+        products_info = pd.DataFrame(products_info_dict["values"].values())
+
+        # MERGE DATA WITH POSITIONS
+        positions_full = pd.merge(positions, products_info, on="id")
+
+        return positions_full
+
+    def fetch_current_positions(self) -> pd.DataFrame:
+        # GET ATTRIBUTES
+        trading_api = self.__trading_api
+
+        # FETCH HELD PRODUCTS
+        request_list = Update.RequestList()
+        request_list.values.extend(
+            [
+                Update.Request(
+                    option=Update.Option.PORTFOLIO,
+                    last_updated=0,
+                ),
+            ]
+        )
+
+        update_pb = trading_api.get_update(
+            request_list=request_list,
+            raw=False,
+        )
+        positions_partial = self.filter_current_positions(
+            portfolio=update_pb.portfolio,
+        )
+
+        # FETCH ADDITIONAL DATA ON PRODUCTS
+        positions = self.fetch_additional_information(
+            positions=positions_partial,
+        )
+
+        return positions
+
+    def help(self):
+        message = (
+            "Degiro:\n"
```

Comment:
You should follow the other menus conventions, i.e.
* Add the help, q, quit commands
* Indentation is only 3 spaces instead of 4
* I would do the other way around with \nDegiro:

I would also put login as being the first, since we're not really following the alphabet but what makes more sense logically let's say